File: rustyCalculator.java
import java.util.Stack;

public class Answer {  
    public static String answer(String str) {
                
        Stack<Character> stack = new Stack<Character>();
        String rpn = "";
        
        for(char c : str.toCharArray()) {
            switch(c) {
            case '+':
                // Better check if there is a higher order operation
                while(!stack.empty() && stack.peek() == '*') {
                    rpn+=stack.pop();
                }
                stack.push(c);
                break;
            case '*':
                stack.push(c);
                break;
            default:
                rpn+=c;
                break;
            }
        }
        
        //All done, let's get our remaining operators
        while(!stack.isEmpty())
            rpn+=stack.pop();
        return rpn;     
    }
}

File: rustyCalculator.py
def answer(str):

    rpn=''
    stack = []

    for c in str:
        if c=='+':
            while len(stack)>0 and stack[len(stack)-1]=='*':
                rpn+=stack.pop()
            stack.append(c)
        elif c=='*':
            stack.append(c)
        else:
            rpn+=(c)
    while len(stack)>0:
        rpn+=stack.pop()
    return rpn

print answer("2+3*2")
print answer("2*4*3+9*3+5")